lib/Dialect/TTIR/Transforms/Passes.cpp

```diff
@@ -573,6 +573,11 @@ class TTIRLayoutDPSOperandsRewriter
     for (auto &operand : op->getOpOperands()) {
       bool isResult = op.isDpsInit(&operand);
 
+      // TTNN Conv2d moves input, weight, and bias from host to device
+      // itself. Inserting the ToLayoutOp on these operands is thus problematic.
+      if (mlir::isa<Conv2dOp>(op.getOperation()) && !isResult) {
+        continue;
+      }
```

Comment:
It seems that there's some gotchas in how tensors are getting to device in general, thru C++ ttnn APIs... I'll chat with Artem to try and understand what the correct way would be to do these things, including the conv op. Can you point me to the code in ttnn I can reference, so that I can better explain what our needs are?

Reply:
If we end up having to treat a couple ops like this, we should probably define a trait and tag those ops with it, should be cleaner that way ("data driven vs code driven").

Reply:
Ideally we'd want the conv op to work if all its inputs are on device already. For conv ops in the middle of a model it would be wasteful to have to move the inputs onto cpu just so the next conv can take them off.